RawMediaAccess/app/src/main/java/com/microsoft/acs/calling/rawmediaaccess/FrameGenerator.java

```diff
@@ -0,0 +1,123 @@
+
+    private VideoFrameSender videoFrameSender;
+    private Thread frameIteratorThread;
+    private final Random random;
+    private volatile boolean stopFrameIterator = false;
+
+    public FrameGenerator() {
+
+        random = new Random();
+    }
+
+    public void FrameIterator() {
+
+        ByteBuffer plane = null;
+        while (!stopFrameIterator && videoFrameSender != null) {
+
+            plane = GenerateFrame(plane);
+        }
+
+        Log.d("FrameGenerator", "FrameGenerator.FrameIterator trace, 1st looper stopped");
+    }
+
+    private ByteBuffer GenerateFrame(ByteBuffer plane) {
+
+        try {
+
+            VideoFormat videoFormat = videoFrameSender.getVideoFormat();
+            if (plane == null || videoFormat.getStride1() * videoFormat.getHeight() != plane.capacity()) {
+
+                plane = ByteBuffer.allocateDirect(videoFormat.getStride1() * videoFormat.getHeight());
+                plane.order(ByteOrder.nativeOrder());
+            }
+
+            int bandsCount = random.nextInt(15) + 1;
+            int bandBegin = 0;
+            int bandThickness = videoFormat.getHeight() * videoFormat.getStride1() / bandsCount;
+
+            for (int i = 0; i < bandsCount; ++i) {
+
+                byte greyValue = (byte) random.nextInt(254);
+                java.util.Arrays.fill(plane.array(), bandBegin, bandBegin + bandThickness, greyValue);
+                bandBegin += bandThickness;
+            }
+
+            SoftwareBasedVideoFrameSender sender = (SoftwareBasedVideoFrameSender) videoFrameSender;
+
+            long timeStamp = sender.getTimestampInTicks();
+            sender.sendFrame(plane, timeStamp).get();
+
+            Thread.sleep((long) (1000.0f / videoFormat.getFramesPerSecond()));
+        } catch (InterruptedException ex) {
+
+            Log.d("FrameGenerator", String.format("FrameGenerator.GenerateFrame, %s", ex.getMessage()));
+            ex.printStackTrace();
+        } catch (ExecutionException ex2) {
+
+            Log.d("FrameGenerator", String.format("FrameGenerator.GenerateFrame, %s", ex2.getMessage()));
+            ex2.getMessage();
+        }
+
+        return plane;
+    }
+
+    private void StartFrameIterator() {
+
+        frameIteratorThread = new Thread(this::FrameIterator);
+        frameIteratorThread.start();
+    }
+
+    public void StopFrameIterator() {
+
+        try {
+
+            if (frameIteratorThread != null) {
+
+                stopFrameIterator = true;
+                Log.d("FrameGenerator", "FrameGenerator.Stop trace, before stopping");
+
+                frameIteratorThread.join();
+                frameIteratorThread = null;
+
+                Log.d("FrameGenerator", "FrameGenerator.Stop trace, after stopping");
+                stopFrameIterator = false;
+            }
+        } catch (InterruptedException ex) {
+
+            Log.d("FrameGenerator", String.format("FrameGenerator.StopFrameIterator, %s", ex.getMessage()));
+            ex.getMessage();
```

Comment:
Unnecessary line, logged on previous one.
```suggestion
            Log.d("FrameGenerator", String.format("FrameGenerator.StopFrameIterator, %s", ex.getMessage()));
```